opencti-platform/opencti-front/src/private/components/entities/cities/Root.tsx

```diff
@@ -0,0 +1,193 @@
+// TODO Remove this when V6
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+import React, { useContext, useMemo } from 'react';
+import { Redirect, Route, Switch, useParams } from 'react-router-dom';
+import { graphql, useLazyLoadQuery, useSubscription } from 'react-relay';
+import { GraphQLSubscriptionConfig } from 'relay-runtime';
+import TopBar from '../../nav/TopBar';
+import City from './City';
+import CityKnowledge from './CityKnowledge';
+import StixDomainObjectHeader from '../../common/stix_domain_objects/StixDomainObjectHeader';
+import FileManager from '../../common/files/FileManager';
+import CityPopover from './CityPopover';
+import StixCoreObjectHistory from '../../common/stix_core_objects/StixCoreObjectHistory';
+import StixCoreObjectOrStixCoreRelationshipContainers
+  from '../../common/containers/StixCoreObjectOrStixCoreRelationshipContainers';
+import StixCoreObjectKnowledgeBar from '../../common/stix_core_objects/StixCoreObjectKnowledgeBar';
+import ErrorNotFound from '../../../../components/ErrorNotFound';
+import EntityStixSightingRelationships from '../../events/stix_sighting_relationships/EntityStixSightingRelationships';
+import { UserContext } from '../../../../utils/Security';
+import { RootCityQuery } from './__generated__/RootCityQuery.graphql';
+import { RootCitiesSubscription } from './__generated__/RootCitiesSubscription.graphql';
+
+const subscription = graphql`
+  subscription RootCitiesSubscription($id: ID!) {
+    stixDomainObject(id: $id) {
+      ... on City {
+        ...City_city
+        ...CityEditionOverview_city
+      }
+      ...FileImportViewer_entity
+      ...FileExportViewer_entity
+      ...FileExternalReferencesViewer_entity
+      ...WorkbenchFileViewer_entity
+    }
+  }
+`;
+
+const cityQuery = graphql`
+  query RootCityQuery($id: String!) {
+    city(id: $id) {
+      id
+      name
+      x_opencti_aliases
+      x_opencti_graph_data
+      ...City_city
+      ...CityKnowledge_city
+      ...FileImportViewer_entity
+      ...FileExportViewer_entity
+      ...FileExternalReferencesViewer_entity
+      ...WorkbenchFileViewer_entity
+    }
+    connectorsForExport {
+      ...FileManager_connectorsExport
+    }
+    settings {
+      platform_enable_reference
+    }
+  }
+`;
+
+const RootCity = () => {
+  const { me } = useContext(UserContext);
+  const { cityId } = useParams() as { cityId: string };
+
+  const link = `/dashboard/entities/cities/${cityId}/knowledge`;
+
+  const subConfig = useMemo<GraphQLSubscriptionConfig<RootCitiesSubscription>>(() => ({
+    subscription,
+    variables: { id: cityId },
+  }), [cityId]);
+  useSubscription(subConfig);
+
+  const data = useLazyLoadQuery<RootCityQuery>(cityQuery, { id: cityId });
```

Comment:
Need to remain the old school way ? Any specific reason?

Reply:
Actually it could be done the "proper" way, but it would require the main Root component to create all the refs of sub-rooting component and send refs. I've not found this very "readable" and chose this path, but we can discuss it.